aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/MutationProcessor.java

```diff
@@ -255,6 +268,14 @@ void stopDrainingMutationOutbox() {
         ));
     }
 
+    private String getModelName(@NonNull Model model) {
+        if (model.getClass() == SerializedModel.class) {
```

Comment:
Feels like we should have got out of the `SerializedModel` business _before_ getting to the mutation processor. It would be nice to have the `SerializeModel` logic contained to a single point in the codebase. Then, we perform the mapping once into the type, once out of the type. The various processing components in the sync engine would not need to do so individually?